sdk/textanalytics/ai-text-analytics/src/util.ts

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { FullOperationResponse, OperationOptions } from "@azure/core-client";
+import { LanguageDetectionInput, TextDocumentInput } from "./generated";
+import { LroResponse } from "@azure/core-lro";
+import { TextAnalysisOperationOptions } from "./models";
+import { logger } from "./logger";
+
+interface IdObject {
+  id: string;
+}
+
+/**
+ * Given a sorted array of input objects (with a unique ID) and an unsorted array of results,
+ * return a sorted array of results.
+ *
+ * @internal
+ * @param sortedArray - An array of entries sorted by `id`
+ * @param unsortedArray - An array of entries that contain `id` but are not sorted
+ */
+export function sortResponseIdObjects<T extends IdObject, U extends IdObject>(
+  sortedArray: T[],
+  unsortedArray: U[]
+): U[] {
+  const unsortedMap = new Map<string, U>();
+  for (const item of unsortedArray) {
+    unsortedMap.set(item.id, item);
+  }
+
+  if (unsortedArray.length !== sortedArray.length) {
+    const ordinal = unsortedArray.length > sortedArray.length ? "more" : "fewer";
+    logger.warning(
+      `The service returned ${ordinal} responses than inputs. Some errors may be treated as fatal.`
+    );
+  }
+
+  const result: U[] = [];
+  for (const sortedItem of sortedArray) {
+    const item = unsortedMap.get(sortedItem.id);
+    if (item) {
+      result.push(item);
+    }
+  }
+  return result;
+}
+
+/**
+ * @internal
+ */
+export interface AssessmentIndex {
+  document: number;
+  sentence: number;
+  assessment: number;
+}
+
+/**
+ * @internal
+ */
+export function parseAssessmentIndex(pointer: string): AssessmentIndex {
```

Comment:
I still keep the dream alive of having a real system for handling JSON pointers in responses.